modules/Emails/EmailsController.php

```diff
@@ -639,4 +673,78 @@ protected function setAfterImport($importedEmailId, $request)
 
         return $emails;
     }
+
+    /**
+     * @param User $requestedUser
+     * @param InboundEmail $requestedInboundEmail
+     * @param Email $requestedEmail
+     * @return bool false if user doesn't have access
+     */
+    protected function userIsAllowedToSendEmail($requestedUser, $requestedInboundEmail, $requestedEmail)
+    {
+        $hasAccess = false;
+
+        // Check that user is allowed to use inbound email account
+        $hasAccessToInboundEmailAccount = false;
+        $usersInboundEmailAccounts = $requestedInboundEmail->retrieveAllByGroupIdWithGroupAccounts($requestedUser->id);
+        foreach ($usersInboundEmailAccounts as $inboundEmailId => $userInboundEmail) {
+            if ($userInboundEmail->id === $requestedInboundEmail->id) {
+                $hasAccessToInboundEmailAccount = true;
+                break;
+            }
+        }
+
+        $inboundEmailStoredOptions = $requestedInboundEmail->getStoredOptions();
+
+        // if group email account check that user is allowed to use email account
+        if ($requestedInboundEmail->isGroupEmailAccount()) {
+            if ($inboundEmailStoredOptions['allow_outbound_group_usage'] === true) {
+                $hasAccessToInboundEmailAccount = true;
+            } else {
+                $hasAccessToInboundEmailAccount = false;
+            }
+        }
+
+        // Check that the from address is the same as the inbound email account
+        $isFromAddressTheSame = false;
+        if ($inboundEmailStoredOptions['from_addr'] === $requestedEmail->from_addr) {
+            $isFromAddressTheSame = true;
+        }
+        // check if user is using the system account, as the email address for the system account
+        // is likely to be different
+        $outboundEmailAccount = new OutboundEmail();
+        if(empty($inboundEmailStoredOptions['outbound_email'])) {
+            $outboundEmailAccount->getUserMailerSettings();
+        } else {
+            $outboundEmailAccount->retrieve($inboundEmailStoredOptions['outbound_email']);
+        }
+
+        $isAllowedToUseOutboundEmail = false;
+        if ($outboundEmailAccount->type === 'system') {
```

Comment:
if the outbound email is null then we will not be able to send an email. As I have previous mentioned in comment above, I meant to fallback to the the system email and then check to see if the user is allowed to use the system account. I have updated my code to better reflect this. Please note that the $isAllowedToUseOutboundEmail is assigned to false. If the id is null it will not match the if conditions which assignthe variable to true. So this will deliberately fail the $hasAccess check.